Site editor sidebar: add page details

SaxonF opened the pull request:
<!-- Thanks for contributing to Gutenberg! Please follow the Gutenberg Contributing Guidelines:
https://github.com/WordPress/gutenberg/blob/trunk/CONTRIBUTING.md -->

## What?
Updates the existing page detail view in the site editor to include more useful information.

## Why?
https://github.com/WordPress/gutenberg/issues/50390

## Constraints

- If we want to get the author and avatar of the user who last modified the page we'll need to fetch and consult the revisions. This is possible, but needs extra work to set up the store in a follow up PR.


## Testing Instructions
1. Activate a block theme
2. Make sure you have some pages ready to role. Add featured images, leave some scheduled too.
3. Now head over to the site editor 
4. In the side bar, navigate to Design > Pages and click on the pages.
5. Check each page's details in the sidebar.
6. Overcome your bedazzlement and provide some constructive feedback. 




<img width="349" alt="Screenshot 2023-05-25 at 3 27 12 pm" src="https://github.com/WordPress/gutenberg/assets/6458278/50d593e8-b11e-4d83-b922-67a58d292506">
<img width="346" alt="Screenshot 2023-05-25 at 3 27 00 pm" src="https://github.com/WordPress/gutenberg/assets/6458278/573a1a31-4813-4a41-9249-afa15b4d9bd6">
<img width="348" alt="Screenshot 2023-05-25 at 3 26 32 pm" src="https://github.com/WordPress/gutenberg/assets/6458278/371f6e29-eb29-463e-b8b1-82d468214d77">

SaxonF committed:
setting the stage for page detail

SaxonF committed:
fix padding issues

ramonjd committed:
Filling in the details :)

jameskoster commented:
Is it a bit strange to see the featured image in the sidebar, if the assigned template doesn't display it? 🤔

ramonjd committed:
This commit:
- exports `getMediaDetails()` from the post-featured-image sidebar component and uses it if there's a featured image in the post record
- cleans up components and unused functions
- uses available wordcount and readtime functions
- implements i18n strings
- shuffles page-specific styles around
- update package log after importing @wordpress/wordcount
- truncating titles and other long excerpts
- adding status icons
- adding featured image description and loading placeholder

ramonjd commented:
I've added a commit 52072770fb1772abb2de909cd7bee292186d5086

Some notes: 

- In order to show last modified by (author) we need to pull the revisions to grab the revision author id, and then match that against a user list to get the display name. This possibly requires new store selectors/actions, so in my opinion it's a potential follow up if required at all.
- The featured image mostly matches the post/page editor display logic, with the exception of forced max-height. If the featured image has a larger height size, then it will be cut off with the risk that it will appear strange. Also, what if the image is not an image? I'm tending to agree with @jameskoster that its inclusion might trigger more raised eyebrows than the feature could possibly contend with. I've left it in this PR to test, but "if in doubt, chuck it out" might apply here.
- Should we bother showing excerpts? They have to be explicitly activated for  pages via `add_post_type_support( 'page', 'excerpt' );`. I've left them in this PR for now just in case.

🍺

ramonjd commented on packages/date/src/test/index.js:
```diff
@@ -642,6 +642,35 @@ describe( 'Moment.js Localization', () => {
+			).toBe( '2 days ago' );
+		} );
+
+		it( 'should return human readable time differences in the future', () => {
```
I just added some tests to make sure it can handle future dates.

ramonjd commented on packages/edit-site/src/components/sidebar-navigation-screen-pages/index.js:
```diff
@@ -28,7 +28,8 @@ const PageItem = ( { postId, ...props } ) => {
 		'postType',
-		'page'
+		'page',
+		{ status: 'any' }
```
Ensures that we show all pages in the list, regardless of status. 

Is this acceptable?

andrewserong replied:
Sounds good to me — we'd want to see all pages in this list, regardless of status, I'd imagine?

ramonjd commented on packages/edit-site/src/components/sidebar-navigation-screen-page/index.js:
```diff
@@ -18,14 +29,124 @@ import SidebarNavigationScreen from '../sidebar-navigation-screen';
+	const wordsCounted = page?.content?.raw
+		? wordCount( page.content.raw, wordCountType )
+		: 0;
+	const readingTime = Math.round( wordsCounted / AVERAGE_READING_RATE );
```
How useful are wordcount and reading time stats to users in this context?

andrewserong replied:
Good question — I quite like the inclusion, it could be useful if you're wanting to see at a glance how long a really long page might be, but I imagine it might be more useful for posts?

jameskoster commented:
@ramonjd The last-modified detail is important, whether here or in a follow-up, as it will serve as an access point to revisions.

I'd say it's nice to include the Featured Image if we can get the appearance right. I can still see folks being a bit confused if they set it, and see no change in the frame. But some help text that clarifies where it can be used would probably address that problem. 

Excerpt seems more useful for posts. And we still need to figure out how editing would work. So I wouldn't mind leaving them out for now. Not a strong feeling though.

<img width="1493" alt="Page" src="https://github.com/WordPress/gutenberg/assets/846565/637d3b31-1f19-40e1-b2ca-bef10488689a">

ramonjd commented:
Thanks for the feedback @jameskoster 

> The last-modified detail is important, whether here or in a follow-up, as it will serve as an access point to revisions.

No worries. Just to confirm, we have the modified date from the post object, but to get the author who made the revision we'll need to fetch the revisions.  

I can look into this separately so it's easier to test. 👍 

> I'd say it's nice to include the Featured Image if we can get the appearance right. I can still see folks being a bit confused if they set it, and see no change in the frame. But some help text that clarifies where it can be used would probably address that problem.

No worries. Let's leave it in. Perhaps we can center the image within the container, but I guess it depends on the container. Not sure how to ensure it will appear 100% of the time. 

> Excerpt seems more useful for posts. And we still need to figure out how editing would work. So I wouldn't mind leaving them out for now. Not a strong feeling though.

It will rarely show anyway so no harm in leaving it in.

jameskoster commented:
> Just to confirm, we have the modified date from the post object, but to get the author who made the revision we'll need to fetch the revisions.

Oh, nice. The _author_ is less important, fine to handle that one separately. 

Here are some outstanding details, and a refreshed mockup.

- The draft status icon looks incorrect, let me know if you need the svg for that. 
- We can probably lay the status row out like the other details, and potentially place it alongside the other details for now. See the latest mockup below.
- Let's add the Featured Image description.
- For the Featured Image position, could we display it as a background image with `background-size: cover;` and `background-position: 50% 50%;`?
- Can we make each detail row 40px tall to match the mockup?
- Ideally the modified date is pinned just above the save panel, and shares the same two column layout as other details.

<img width="1560" alt="Page" src="https://github.com/WordPress/gutenberg/assets/846565/b10fa452-3761-42ef-a677-9ea09715bd51">

Thanks! I'm try to handle some of the other styling details when we get a bit closer.

ramonjd commented:
> The draft status icon looks incorrect, let me know if you need the svg for that.

Oh, I didn't realize there were mock ups. I'm looking at this [screenshot](https://github.com/WordPress/gutenberg/issues/50390#issuecomment-1553163817). I'll see how far I get with CSS + icons, but it might be useful to have SVGs if I run into geometrical challenges. Cheers!

> Ideally the modified date is pinned just above the save panel, and shares the same two column layout as other details.

I tried that yesterday. It might entail some restructuring of the HTML. Will have another crack.

_Edit_: I can't find a neat way to do this within the constraints of `<SidebarNavigationScreen />` that doesn't cause the modified date container to overlap the other details when the page height is reduced. The only reliable way I think is to create a slot in `<Sidebar />` inside the `.edit-site-sidebar__footer` element, similar to what we've done for the "Apply" button on Global styles revisions.

For now I've stretched the flex item and rendering the modified date link beneath the items but with a bit of padding for good luck.

jameskoster commented:
It might be worth looking at how the 'dynamic pages' group is handled in https://github.com/WordPress/gutenberg/pull/50630 for implementation? It's basically the same thing. 

Here are the svgs:

**Published**

```
<svg fill="none" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 16 16">
  <path fill-rule="evenodd" clip-rule="evenodd" d="M8 16A8 8 0 1 0 8 0a8 8 0 0 0 0 16Zm-1.067-5.6 4.2-5.667.8.6-4.8 6.467-3-2.267.6-.8 2.2 1.667Z"/>
</svg>
```

**Draft**
```
<svg fill="none" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 16 16">
  <path fill-rule="evenodd" clip-rule="evenodd" d="M14.5 8a6.5 6.5 0 1 1-13 0 6.5 6.5 0 0 1 13 0ZM16 8A8 8 0 1 1 0 8a8 8 0 0 1 16 0Zm-8 4a4 4 0 0 0 0-8v8Z"/>
</svg>
```

**Pending / Scheduled**
These are the same, just different colors.

```
<svg fill="none" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 16 16">
  <path fill-rule="evenodd" clip-rule="evenodd" d="M14.5 8a6.5 6.5 0 1 1-13 0 6.5 6.5 0 0 1 13 0ZM16 8A8 8 0 1 1 0 8a8 8 0 0 1 16 0ZM8 4a4 4 0 1 0 0 8 4 4 0 0 0 0-8Z"/>
</svg>
```

ramonjd commented:
Great, thanks for the PNGs. Nice!

> It might be worth looking at how the 'dynamic pages' group is handled in https://github.com/WordPress/gutenberg/pull/50630 for implementation? It's basically the same thing.

It which sense? The [sticky footer](https://github.com/WordPress/gutenberg/pull/50630/files#diff-2fb742b90228e5b66e14e3b70aa34b47d38de61c9bebcddb205537dab7a4935bR2)? I'll have a closer look and see if it's possible in this case. I'd submit it isn't a blocker for MVP.


_Edit_: Looks like it implements something I already tried and cast aside because it overlaps the details as the window height reduces - if you're happy with that I'll reinstate. At least things will be consistent and we can always tweak later. Only sticking the element to the bottom of an 100% container and catering for resize + scroll was tripping me up. Cheers again!! 🙇

ramonjd committed:
Adding SVG icons for status labels

Removing duped/unused test expectations
De-nesting BEM rules
Remove unused comments
i18n

andrewserong commented on packages/date/src/test/index.js:
```diff
@@ -642,6 +642,35 @@ describe( 'Moment.js Localization', () => {
 				)
 			).toBe( '2 days ago' );
+
+			expect(
+				humanTimeDiff(
+					'2023-04-28T11:00:00.000Z',
+					'2023-04-28T13:00:00.000Z'
+				)
+			).toBe( '2 hours ago' );
+
+			expect(
+				humanTimeDiff(
+					'2023-04-26T13:00:00.000Z',
+					'2023-04-28T13:00:00.000Z'
+				)
+			).toBe( '2 days ago' );
```
Are these two effectively the same as the above two `expect` statements?

ramonjd replied:
Whoops, good pick up. That was a leftover from hacking.

andrewserong commented on packages/edit-site/src/components/sidebar-navigation-data-list/style.scss:
```diff
@@ -0,0 +1,7 @@
+	color: $gray-600;
+}
+
+.edit-site-sidebar-navigation_data-list__item .edit-site-sidebar-navigation_data-list__value {
```
Tiny nit: since this uses a BEM naming convention, is the parent `.edit-site-sidebar-navigation_data-list__item` needed? Could this just be a single `.edit-site-sidebar-navigation_data-list__value` line, or does it need higher specificity?

```suggestion
.edit-site-sidebar-navigation_data-list__value {
```

ramonjd replied:
This one is for specificity. I can change it to `.edit-site-sidebar-navigation_data-list__value.edit-site-sidebar-navigation_data-list__value` maybe to communicate that. 👍

andrewserong commented on packages/edit-site/src/components/sidebar-navigation-screen-page/index.js:
```diff
@@ -18,40 +34,238 @@ import SidebarNavigationScreen from '../sidebar-navigation-screen';
+
+	if ( page?.status ) {
+		details.push( {
+			label: 'Status',
```
Should each of these `label` values be wrapper in translation (`__()`) calls?

ramonjd replied:
Yes! They should! 👍

andrewserong commented on packages/edit-site/src/components/sidebar-navigation-screen-page/style.scss:
```diff
@@ -0,0 +1,66 @@
+	background-color: $gray-800;
+	margin-bottom: $grid-unit-20;
+	min-height: 128px;
+	.edit-site-sidebar-navigation-screen-page__featured-image {
```
Similar nit to another comment, is the nesting here needed?

ramonjd replied:
Not here! 🙇

andrewserong reviewed (commented):
This is looking great to me, nice work folks!

> Overcome your bedazzlement and provide some constructive feedback.

I was mostly bedazzled — it's looking very cool indeed, but I left a few nits. The only issue I ran into is that it looks like some of the labels might need to be wrapped in translate calls, unless I'm missing something?

Visually it's looking and testing great for me 🎉 

| Published | Draft with image | Scheduled with parent |
| --- | --- | --- |
| ![image](https://github.com/WordPress/gutenberg/assets/14988353/395aa782-3686-41b9-a902-22a3df3916c2) | ![image](https://github.com/WordPress/gutenberg/assets/14988353/b7e59c43-2875-49a0-beb0-a976128270c0) | ![image](https://github.com/WordPress/gutenberg/assets/14988353/4ccd6b17-770c-44e1-b040-bd295dba9d5c) |

andrewserong commented on packages/edit-site/src/components/sidebar-navigation-data-list/index.js:
```diff
@@ -0,0 +1,36 @@
+	__experimentalText as Text,
+} from '@wordpress/components';
+
+/**
+ * Internal dependencies
+ */
```
Nit: stray comment that can be removed?

ramonjd commented:
RTL

<img width="349" alt="Screenshot 2023-05-25 at 2 43 12 pm" src="https://github.com/WordPress/gutenberg/assets/6458278/b1e9f649-7286-4d8e-8633-415a9a41f329">

andrewserong commented on packages/edit-site/src/components/sidebar-navigation-screen-page/index.js:
```diff
@@ -18,40 +34,238 @@ import SidebarNavigationScreen from '../sidebar-navigation-screen';
+	if ( wordsCounted ) {
+		details.push(
+			{
+				label: 'Words',
+				value: wordsCounted.toLocaleString() || __( 'Unknown' ),
+			},
+			{
+				label: 'Time to read',
```
Just spotted these labels for translation, too 🙂

ramonjd committed:
Removing duped/unused test expectations
De-nesting BEM rules
Remove unused comments
i18n

andrewserong reviewed (approved):
Thanks for all the updates!

This is still testing well for me, and I also double-checked that it displays nicely when certain fields aren't available, e.g. for a page with no words, both the `Words` and `Time to read` rows disappear:

<img width="350" alt="image" src="https://github.com/WordPress/gutenberg/assets/14988353/fa8ac312-cb8d-4f74-9c24-90bce4e78f48">

Code-wise this LGTM! ✨

I haven't been very involved in the discussion about this particular part of the UI, so you might want to get a final design review, too 🙂

ramonjd committed:
Displaying templates

ramonjd commented:
Thanks for the review @andrewserong 🙇

ramonjd committed:
We no longer need to pass down a className to SidebarNavigationScreen. Reverting.

SaxonF committed:
style changes page detail site editor

jameskoster committed:
Update colors

jameskoster commented:
> I'd submit it isn't a blocker for MVP.

Yep, not a blocker. We can handle it separately as it affects other panels (See the "Footer" section in https://github.com/WordPress/gutenberg/issues/50379).

I updated a couple of colors based on the latest designs. 

One other detail to note, unsure if it needs to be addressed here, but the browser tab title is a little strange. I see: 

> Template name < Template < Editor < Site Title 

Should we include the Page title there? Perhaps something like 

> Page title < Pages < Editor < Site Title

Finally, I'm not seeing the animation on the frame when moving between details / edit:

https://github.com/WordPress/gutenberg/assets/846565/83f4f0ac-2902-46f5-97de-0d4d1449add6

Compared to Trunk:

https://github.com/WordPress/gutenberg/assets/846565/aef7e304-2775-4889-abea-789f0a5462c4

ntsekouras commented on packages/edit-site/src/components/sidebar-navigation-data-list/index.js:
```diff
@@ -0,0 +1,25 @@
+ */
+import DataListItem from './data-list-item';
+
+export default function SidebarDetails( { details } ) {
```
The naming of the file and the above `DataListItem` component feel a bit ambiguous to me. Are we going to reuse them elsewhere? Should we move them internally to `SidebarNavigationScreenPage`?

ramonjd replied:
Agree. I suspect it's specific to display individual entity properties, e.g., page/template. @SaxonF can confirm.

We can rename in a follow up if so.

ntsekouras commented:
Thanks for the PR! Is this how it's supposed to look or is it something with my local cache or something? 😄 
I'm saying that because @andrewserong screenshot has different alignments(right).

<img width="354" alt="Screenshot 2023-05-25 at 5 15 49 PM" src="https://github.com/WordPress/gutenberg/assets/16275880/247facbc-e125-4b1c-a4ff-f83bf894c3fc">

ntsekouras commented on packages/edit-site/src/components/sidebar-navigation-screen-page/status-label.js:
```diff
@@ -0,0 +1,98 @@
+			statusLabel = __( 'Pending' );
+			break;
+		default:
+			statusLabel = __( 'Unknown' );
```
Which case would end up here?

ramonjd replied:
Oh, good spotting. This was a leftover from a previous version. I think we can remove since we're already checking for `record?.status` before rendering the detail.

ntsekouras commented on packages/editor/src/components/post-featured-image/index.js:
```diff
@@ -43,7 +43,7 @@ const instructions = (
 );
 
-function getMediaDetails( media, postId ) {
+export function getMediaDetails( media, postId ) {
```
This is specific function for featured image and the name is generic, because it was used internally by that component. I think it would be better to `lock` this an make it private instead of introducing a new API.

ramonjd replied:
Good call! Thanks for the nudge.

ntsekouras commented on packages/edit-site/src/components/sidebar-navigation-screen/index.js:
```diff
@@ -40,7 +42,7 @@ export default function SidebarNavigationScreen( {
 
 	return (
-		<VStack spacing={ 2 }>
+		<VStack spacing={ 0 }>
```
This affects many screen, right? Just to confirm we want the design updated in this component.

ramonjd replied:
I believe this is intentional yes. 

cc @SaxonF

ntsekouras commented on packages/edit-site/src/components/sidebar-navigation-screen-page/index.js:
```diff
@@ -18,6 +33,78 @@ import SidebarNavigationScreen from '../sidebar-navigation-screen';
+	}
+	const details = [];
+
+	if ( page?.status ) {
```
Can we have pages without `status` or `slug`?

ramonjd replied:
In my experience, no, but I wouldn't expect it hurts to be defensive. Happy to remove these checks since we're already checking if the record is not `undefined` (which it is until the selector resolves)

ntsekouras commented on packages/edit-site/src/components/sidebar-navigation-screen-page/index.js:
```diff
@@ -18,6 +33,78 @@ import SidebarNavigationScreen from '../sidebar-navigation-screen';
+	 * Do not translate into your own language.
+	 */
+	const wordCountType = _x( 'words', 'Word count type. Do not translate!' );
+	const wordsCounted = page?.content?.raw
```
I think we should use the `rendered` content and not the `raw` one(shortcodes etc..).

ntsekouras commented on packages/edit-site/src/components/sidebar-navigation-screen-page/index.js:
```diff
@@ -26,34 +113,158 @@ export default function SidebarNavigationScreenPage() {
+
+			// Parent page title.
+			const parent = record?.parent
+				? select( coreStore ).getEntityRecord(
```
Nit: we can probably retrieve only the title here(`fields`)

ntsekouras commented on packages/edit-site/src/components/sidebar-navigation-screen-page/index.js:
```diff
@@ -26,34 +113,158 @@ export default function SidebarNavigationScreenPage() {
+							) }
+						</div>
+					</VStack>
+					{ !! record?.excerpt?.raw && (
```
Same comment regarding `raw` vs `rendered`.

ramonjd replied:
We'll have to use `stripHTML` instead because it comes with `<p>` tags. `dangerouslySetInnerHTML` doesn't appear to work with `<Truncate />`

ntsekouras commented on packages/edit-site/src/components/sidebar-navigation-screen-page/index.js:
```diff
@@ -26,34 +113,158 @@ export default function SidebarNavigationScreenPage() {
+						Details
+					</SidebarNavigationSubtitle>
+					<SidebarDetails
+						details={ getPageDetails( {
```
I feel all these info should be inside the `SidebarDetails`. I can't see a reason we should have them in the top component.

ramonjd replied:
I think this is also deliberate since  `<SidebarDetails />` was to be used for single template panels as well? 

🤷 Not sure!

cc @SaxonF for confirmation

SaxonF replied:
Yeah `SidebarDetails` is just a generic data list component we can re-use across multiple panels , including templates and template parts.

ntsekouras replied:
Still, the `page details` data could be isolated in a hook or in the details component..

ramonjd committed:
Ensuring that getFeaturedMediaDetails can be used locally.
Importing getFeaturedMediaDetails into privateApis for the editor package.
Removing unnecessary conditional checks
Escaping HTML and other values for rendering in page and attributes - adding escape-html to package.json
Don't render details when we don't have a record

ramonjd commented:
Thanks for reviewing @ntsekouras ! I've made those changes and rebased.

I think this should be good for a final test and then 🚀

andrewserong commented on packages/edit-site/src/components/sidebar-navigation-screen-page/status-label.js:
```diff
@@ -0,0 +1,95 @@
+			break;
+	}
+
+	return (
+		<div
+			className={ classnames(
+				'edit-site-sidebar-navigation-screen-page__status',
+				{
+					[ `has-status has-${ status }-status` ]: !! status,
+				}
+			) }
+		>
+			{ statusIcon } { statusLabel }
+		</div>
+	);
+}
```
It appears we render an empty Status area if a page is set to "private":

<img width="348" alt="image" src="https://github.com/WordPress/gutenberg/assets/14988353/71901089-5bda-4248-b8c9-f755d3de6849">

Should we add a label for private status? And if there's no match for status, would it be better to hide the whole row, or leave blank as it is currently?

ramonjd replied:
I believe we should hide the whole row if we determine a status, thanks for confirming this. I'll update.

ramonjd replied:
Actually 'private' is a status, but we don't cater for it in terms of icon or copy

ramonjd replied:
Also, what about password protected? We can detect that as well. Perhaps we need more icons?  

I've done this for now:

<img width="323" alt="Screenshot 2023-05-26 at 2 02 46 pm" src="https://github.com/WordPress/gutenberg/assets/6458278/ead32940-2fdb-4cc2-990d-6018b483a62d">


<img width="335" alt="Screenshot 2023-05-26 at 2 02 56 pm" src="https://github.com/WordPress/gutenberg/assets/6458278/7d21cde6-9956-4511-8d26-4a508f35447d">

andrewserong replied:
Yeah, for private posts/pages, I was imagining it could just be a `Private` status label with no icon, a little bit like the `pending` status for now.

But also, feel free to leave it for a follow-up — this PR is already fairly large, so it could be easier to fine-tune / fix up these things in follow-ups if you'd prefer. Whichever works best for you!

andrewserong replied:
Sorry, missed your reply before commenting. Those work nicely for me!

ramonjd replied:
Thanks! Icons are easy to add in a follow up. Let's merge this!

andrewserong reviewed (commented):
This is still testing well for me! Just found a tiny issue that private pages shows up with an empty status field, so left a comment there.

> Thanks for the PR! Is this how it's supposed to look or is it something with my local cache or something? 😄
I'm saying that because @andrewserong screenshot has different alignments(right).

It looks like the designs were updated since I last reviewed, and the PR appears to be consistent with the designs shared in https://github.com/WordPress/gutenberg/issues/50390. I think personally, I slightly prefer the right aligned look, but don't feel too strongly about it.

One potential issue is with how each of the value fields look when the data inside is wrapped. To my eyes the scheduled date in the following looks a little odd because it's center aligned vertically, so the value is visually higher than the detail label:

<img width="332" alt="image" src="https://github.com/WordPress/gutenberg/assets/14988353/9cf0ebba-003a-4ffb-b359-e14aecd787a3">

This is quite fine-tuning stuff, though, so personally I wouldn't think either of the above issues as blockers to landing if it's easier to get the larger PR in, and then tweak things in follow-ups.

ramonjd commented:
> To my eyes the scheduled date in the following looks a little odd because its center aligned vertically, so the value is visually higher than the detail label:

I'll leave this to @jameskoster or @SaxonF  😄

ramonjd committed:
Let's show password protected and private page status as well.

ramonjd commented:
I think we should merge before this PR gets too hairy. Follow ups will be a cinch. 🍺

andrewserong commented:
> I think we should merge before this PR gets too hairy. 

Agreed. Nice work on this PR! 🎉 :shipit:

ramonjd commented:
> Finally, I'm not seeing the animation on the frame when moving between details / edit:

I'm not sure it's related to this PR as it's pretty contained in the page panel. I can't replicate?
![2023-05-26 14 57 20](https://github.com/WordPress/gutenberg/assets/6458278/073d2fda-f141-4fb8-b2b5-984ede124bef)

youknowriad commented on packages/editor/src/components/post-featured-image/get-featured-media-details.js:
```diff
@@ -0,0 +1,64 @@
+ * @param {number} postId The post ID of the media
+ * @return {MediaDetails} The featured media details.
+ */
+export default function getFeaturedMediaDetails( media, postId ) {
```
It is very unclear to me why this is a selector of "editor" package instead of being a "hook" in edit-site for instance. `editor` should never be imported in edit-site IMO.

youknowriad replied:
Looking further at the PR, I guess we want to reuse the same functionality that was present in this component. I think "core-data" is a more suited place for this and I'm also not sure whether it should be a hook or selector.

ramonjd replied:
There is the option to revert this entirely and use the medium image size, and where that's not available the full one in the sidebar. 

The image dimensions are constrained and cropped by CSS anyway so, image download size aside, the width and height are not as important since we're not displaying the entire image as we are in the block inspector.

youknowriad replied:
> There is the option to revert this entirely and use the medium image size, and where that's not available the full one in the sidebar.

Yeah I can't tell why we had the logic in the sidebar in the first place, it's also a weird logic: runs a filter using the "large" size, otherwise run the same filter using the "thumbnail" size. There's probably some historic reasons for these filters to be required.

youknowriad commented on packages/edit-site/src/components/sidebar-navigation-screen-page/index.js:
```diff
@@ -18,42 +35,235 @@ import SidebarNavigationScreen from '../sidebar-navigation-screen';
+			return {
+				parentTitle: _parentTitle,
+				templateSlug: getEditedPostContext()?.templateSlug,
+				featuredMediaDetails: {
```
This component is going to be re-rendered on each store change. We shouldn't return random objects from useSelect callbacks. We should have a lint rule about that ideally.

youknowriad commented on packages/edit-site/src/components/sidebar-navigation-screen-page/index.js:
```diff
@@ -18,42 +35,235 @@ import SidebarNavigationScreen from '../sidebar-navigation-screen';
 
 export default function SidebarNavigationScreenPage() {
 	const { setCanvasMode } = unlock( useDispatch( editSiteStore ) );
+	const { getFeaturedMediaDetails } = unlock( privateEditorApis );
```
Why is this called within the component instead of top level?

youknowriad commented on packages/edit-site/src/components/sidebar-navigation-screen-page/index.js:
```diff
@@ -18,42 +35,235 @@ import SidebarNavigationScreen from '../sidebar-navigation-screen';
+					? decodeEntities( _parentTitle.title.rendered )
+					: __( 'Untitled' );
+			} else {
+				_parentTitle = __( 'Top level' );
```
I'm not sure returning translated strings within useSelect is a good idea? Not sure if we've done this in the past. Translations feel more like "leaf" things and they can be filtered as well.

youknowriad commented on packages/edit-site/src/components/sidebar-navigation-screen-page/index.js:
```diff
@@ -18,42 +35,235 @@ import SidebarNavigationScreen from '../sidebar-navigation-screen';
+	const { records: templates, isResolving: areTemplatesLoading } =
+		useEntityRecords( 'postType', 'wp_template', {
+			per_page: -1,
+		} );
```
Do we really need to fetch all the templates, to retrieve one by slug, can't we call a single `getEntityRecord` instead?

ntsekouras replied:
I had checked this too. We cannot make a request for templates by `slug` if I'm not wrong. That's something we should address though at some point.

youknowriad replied:
I think we can, for templates the `theme//slug` is the id

ramonjd replied:
I also tried `getEntityRecord` and it does return `slug`, but only the template name segment, e.g., 

```js
{
    "postType": "page",
    "postId": "2",
    "templateSlug": "page"
}
```

So I needed a way to get the current theme to build the id, e.g., `twentytwentythree//${ _templateSlug }`

ramonjd replied:
And now I've just discovered `getCurrentTheme()` :trollface:

youknowriad replied:
It might be good to check how we're retrieving the "post/page" template in the sidebar of the post editor. It's the same issue here.

ramonjd replied:
Nice! 

That's the first place I looked but it's using `availableTemplates`, which is a property of `select( editorStore ).getEditorSettings()`, but now I see the site editor settings have it under `defaultTemplateTypes`. 👍 Updated.

youknowriad commented on packages/edit-site/src/components/sidebar-navigation-screen-page/index.js:
```diff
@@ -18,42 +35,235 @@ import SidebarNavigationScreen from '../sidebar-navigation-screen';
+						Details
+					</SidebarNavigationSubtitle>
+					<SidebarDetails
+						details={ getPageDetails( {
```
This function is a bit weird, Can't we instead transform it to a `<PageDetails id={pageId}  />` instead and collocate all the logic?

youknowriad commented on packages/edit-site/src/components/sidebar-navigation-screen-page/status-label.js:
```diff
@@ -0,0 +1,101 @@
+			d="M14.5 8a6.5 6.5 0 1 1-13 0 6.5 6.5 0 0 1 13 0ZM16 8A8 8 0 1 1 0 8a8 8 0 0 1 16 0ZM8 4a4 4 0 1 0 0 8 4 4 0 0 0 0-8Z"
+		/>
+	</SVG>
+);
```
@jasmussen Are we ok with all these adhoc icons? Should we add them to the icons library?

ramonjd replied:
They're specific to the page details, but happy to move them to the icons library. I have a refactor PR going at https://github.com/WordPress/gutenberg/pull/51093

youknowriad commented on packages/edit-site/src/components/sidebar-navigation-screen-page/status-label.js:
```diff
@@ -0,0 +1,101 @@
+	</SVG>
+);
+
+export default function StatusLabel( { status, date } ) {
```
Should we extract this one to its own file/folder? Feels more like a reusable UI component?

ramonjd replied:
I think we could, but maybe should wait until there's a use case? At the moment it's 100% specific to pages so I reckon we could keep in contained within the [suggested PageDetails](https://github.com/WordPress/gutenberg/pull/50767/files#r1209818178) until there's a need?

youknowriad replied:
I'm not saying extract it to the "components" package yet but the file looks a bit crowded to me. But it's minor, fine to wait.

youknowriad reviewed (commented):
Personally I feel this PR has been merged a little bit too quickly but it's fine we have time to correct. I've left a few comments here.

ramonjd commented:
>  I've left a few comments here.

Thanks for looking at it @youknowriad

I'll make a follow up tomorrow to address your feedback.

ramonjd commented:
First stab at refactor PR over at https://github.com/WordPress/gutenberg/pull/51093
